### PR TITLE
Feature bot 590 datalayer for traffic repport

### PR DESCRIPTION
**Pull request by tchevily**
:warning: This PR depends on https://github.com/CanalTP/SamEcoreUserManagerBundle/pull/26

# Description
This PR pushes TR user datas in datalayer

## Pull Request type

This is new feature

## How to test

Here are the following steps to test this pull request:

- Check exposed data in HTML code source when user isn't authenticated
- Check exposed data in HTML code source when user is authenticated

### CODE REVIEW

File: Resources/views/datalayer_user.html.twig [hunk 1]
```diff
@@ -1,8 +1,55 @@
...
+
+    {% set clintUsersNb = app.user.customer.users|length %}
+
+    {% for app in app.user.customer.applications %}
+        {% if app.application.canonicalName == 'realtime' %}{% set userHasRightsOnTR = 'oui' %}{% endif %}
+    {% endfor %}
+
+    {% if userHasRightsOnTR == 'oui' %}
+        {% for r in app.user.userRoles %}
+            {% set trUserRoles = trUserRoles|merge([r.name|replace({'RealTime':'', ' ': ''})]) %}
```
Comment (pthegner) [inline Resources/views/datalayer_user.html.twig]: I think it's only the role for TR part not all the role of the user.

File: Resources/views/datalayer_user.html.twig [hunk 2]
```diff
@@ -1,8 +1,55 @@
...
+
+{% set dataLayerData = {
+        'user_logged': userIsLogged,
+        'client_id': app.user.customer.id|default('unknown'),
+        'client_user_nr': clintUsersNb,
+        'user_id': app.user.id|default('unknown'),
+        'user_TR_rights' : userHasRightsOnTR,
+        'TR_user_role' : trUserRoles|join('/'),
+        'TR_user_creation_date' : trUserCreatedAt,
+        'TR_user_validation_date' : trUserExpiresAt
```
Comment (pthegner) [inline Resources/views/datalayer_user.html.twig]: I don't think that TR_user_validation_date is the date of expiration

File: Resources/views/datalayer_user.html.twig [hunk 3]
```diff
@@ -1,8 +1,55 @@
...
+
+    {% endif %}
+
+{% endif %}
+
+
+{% set dataLayerData = {
+        'user_logged': userIsLogged,
+        'client_id': app.user.customer.id|default('unknown'),
+        'client_user_nr': clintUsersNb,
```
[the remark this answers is not in the document]
Comment (pthegner) [reply Resources/views/datalayer_user.html.twig]: no @Netmisa, if you check BOT-590 and the doc from BOT-591, you should see that it is really client_user_nr (maybe it's the first and last letter of "number")

File: Resources/views/datalayer_user.html.twig [hunk 4]
```diff
@@ -1,8 +1,55 @@
...
+        {% for r in app.user.userRoles %}
+            {% set trUserRoles = trUserRoles|merge([r.name|replace({'RealTime':'', ' ': ''})]) %}
+        {% endfor %}
+
+        {% set trUserExpiresAt = app.user.expiresAt %}
+        {% if trUserExpiresAt %}
+            {% set trUserExpiresAt = app.user.expiresAt|date("d/m/Y") %}
+        {% endif %}
+
+        {% set trUserCreatedAt = app.user.createdAt %}
```
[the remark this answers is not in the document]
Comment (pthegner) [reply Resources/views/datalayer_user.html.twig]: @Netmisa, see BOT-590 for an answer

File: Resources/views/datalayer_user.html.twig (hunk 3 above)
[the remark this answers is not in the document]
Comment (Netmisa) [reply Resources/views/datalayer_user.html.twig]: I think it's mistake in BOT-590 ...
You see @tchevily use `clientUsersNb` because it's more clear

**Pull request merged**